pkg/cloudprovider/providers/azure/azure_managedDiskController.go

```diff
@@ -40,56 +40,94 @@ type ManagedDiskController struct {
 	common *controllerCommon
 }
 
+// ManagedDiskOptions specifies the options of managed disks.
+type ManagedDiskOptions struct {
+	DiskName           string
+	SizeGB             int
+	PVCName            string
+	ResourceGroup      string
+	Zoned              bool
+	ZonePresent        bool
```

Comment:
Repeating my comment here so it shows up on the bottom.

Consider only supporting a zones parameter, and not zone.  We only did that in aws and gce for backwards compatibility reasons, but since all of this is new, you could simplify the code a little bit from the start.

Reply:
@msau42's suggestion makes sense, only one `zones` parameter would make the design much more clear, would you address that in this PR or next? @feiskyer